### PR TITLE
Handle k3d cluster vs dockerized Busola

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Handle DOCKER_DESKTOP_CLUSTER params - rewrite `0.0.0.0` address to `host.docker.internal`.
- Get rid of deprecated `url.parse` - just use WHATWG URL API .
- Update docs.
- Small personal contribution - allow custom namespace in `resources/apply-resources.sh`.

**Testing**

Use the `wawrzyn321/busola-local` image.

**Related issue(s)**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+
+## Busola in Docker against a k3d cluster
+
+Due to the fact the k3d cluster's API server is exposed on `0.0.0.0` address, connectivity problems may occur.
+
+- For Docker Desktop for Mac and Windows, pass `DOCKER_DESKTOP_CLUSTER=true` on dockerized Busola startup.
+
+```bash
+docker run -p 3001:3001 -e DOCKER_DESKTOP_CLUSTER=true busola/local:latest
+```
```
Comment (NHingerl) [inline README.md:93]:
```suggestion
   ```
```

File: README.md [hunk 2]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Busola in Docker against a k3d cluster
```
Comment (NHingerl) [inline README.md:85]:
```suggestion
## Symptom
You are experiencing connectivity problems with Busola in Docker against a k3d cluster.
```

We are trying to establish a fixed headline structure for troubleshooting topics :)

File: README.md [hunk 3]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+
+Due to the fact the k3d cluster's API server is exposed on `0.0.0.0` address, connectivity problems may occur.
+
```
Comment (NHingerl) [inline README.md:88]:
```suggestion
## Remedy
```

File: README.md [hunk 4]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+- For Docker Desktop for Mac and Windows, pass `DOCKER_DESKTOP_CLUSTER=true` on dockerized Busola startup.
```
Comment (NHingerl) [inline README.md:89]:
```suggestion
- For Docker Desktop for Mac and Windows, pass `DOCKER_DESKTOP_CLUSTER=true` on dockerized Busola startup. This way, `0.0.0.0` is automatically replaced with `host.docker.internal`.
```

File: README.md [hunk 5]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+
+```bash
+docker run -p 3001:3001 -e DOCKER_DESKTOP_CLUSTER=true busola/local:latest
+```
+
+`0.0.0.0` will be automatically replaced with `host.docker.internal`.
```
Comment (NHingerl) [inline README.md:95]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+
+- For Linux, run Busola with `--net=host` (omitting the `-p` parameter).
+
+```bash
```
Comment (NHingerl) [inline README.md:99]:
```suggestion
   ```bash
```

File: README.md [hunk 7]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+docker run --net=host busola/local:latest
```
Comment (NHingerl) [inline README.md:100]:
```suggestion
   docker run --net=host busola/local:latest
```

File: README.md [hunk 8]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+```
```
Comment (NHingerl) [inline README.md:101]:
```suggestion
   ```
```

**Review by NHingerl — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Wawrzyn321) [reply README.md:93]: Wait, but this way all text after this block will be rendered as a bash script...?
image

File: README.md [hunk 9]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Busola in Docker against a k3d cluster
+
+Due to the fact the k3d cluster's API server is exposed on `0.0.0.0` address, connectivity problems may occur.
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply README.md:88]: I've added something more specific, I hope it's fine now.

**Review by valentinvieriu — DISMISSED** [review]

File: README.md [hunk 10]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Symptom
+
+You are experiencing connectivity problems with Busola in Docker against a k3d cluster.
+## Remedy
+When the k3d cluster's API server is exposed on `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to wrong endpoint.
```
Comment (NHingerl) [inline README.md:89]:
```suggestion
When the k3d cluster's API server is exposed on the `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to wrong endpoint.
```

File: README.md [hunk 11]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Symptom
+
+You are experiencing connectivity problems with Busola in Docker against a k3d cluster.
+## Remedy
```
Comment (NHingerl) [inline README.md:88]:
```suggestion
## Cause
```
That's the cause of the issue, not yet the remedy

File: README.md [hunk 12]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+When the k3d cluster's API server is exposed on `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to wrong endpoint.
+
```
Comment (NHingerl) [inline README.md:90]:
```suggestion
## Remedy
```
This is where the remedy instructions start :)

**Review by NHingerl — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Symptom
+
+You are experiencing connectivity problems with Busola in Docker against a k3d cluster.
+## Cause
+When the k3d cluster's API server is exposed on the `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to wrong endpoint.
```
Comment (NHingerl) [inline README.md:89]:
```suggestion
When the k3d cluster's API server is exposed on the `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to the wrong endpoint.
```

File: README.md [hunk 14]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
 ## Troubleshooting
 
 > **TIP:** To solve most of the problems with Busola development, clear the browser cache or do a hard refresh of the website.
+
+## Symptom
+
+You are experiencing connectivity problems with Busola in Docker against a k3d cluster.
+## Cause
```
Comment (NHingerl) [inline README.md:88]:
```suggestion

## Cause
```

File: README.md [hunk 15]
```diff
@@ -81,3 +81,21 @@ For the information on how to run tests and configure them, go to the [`tests`](
...
+When the k3d cluster's API server is exposed on the `0.0.0.0` address on you machine, Busola in Docker interprets `0.0.0.0` as its internal Docker address, routing the requests to wrong endpoint.
+## Remedy
```
Comment (NHingerl) [inline README.md:90]:
```suggestion

## Remedy
```

**Review by NHingerl — CHANGES_REQUESTED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by valentinvieriu — APPROVED** [review]

**Review by akucharska — APPROVED** [review]

**Pull request merged**